homeassistant/components/holiday/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow for Holiday integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from babel import Locale
+from holidays import list_supported_countries
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    CountrySelector,
+    CountrySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_PROVINCE, DOMAIN
+
+SUPPORTED_COUNTRIES = list_supported_countries(include_aliases=False)
+
+
+class HolidayConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Holiday."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is not None:
+            self.data = user_input
+
+            selected_country = self.data[CONF_COUNTRY]
```

Comment:
Always prefer using the local variable over the attribute when possible. Here we can use `user_input`.